### PR TITLE
Fixes #613.

### LINKED ISSUE #613: NRE when auto-connecting to a ServiceControl instance that is no longer available
Steps to reproduce:

1. Start SI and connect to an instance of SC
2. Exit SI
3. Stop or remove that instance 
4. Start SI

Upon startup SI tries to connect to the previously connected instance, but it throws an exception. The application crashes on Splash screen.

### PR DESCRIPTION
**Pull request by HEskandari**
Doing a null check on the url SI connects to. Also doing that on OnInitialize to fix the auto-connect on startup issue.

### CODE REVIEW

**Comment by HEskandari** [comment]
@Particular/serviceinsight-maintainers Please review.

**Comment by adamralph** [comment]
@HEskandari why was this tagged as being blocked?

File: src/ServiceInsight/ServiceControl/ServiceControlConnectionProvider.cs [hunk 1]
```diff
@@ -15,6 +15,11 @@ public void ConnectTo(string url)
         {
             Url = url;
 
+            if (url == null)
```
Comment (adamralph) [inline src/ServiceInsight/ServiceControl/ServiceControlConnectionProvider.cs]: Would it be better to throw an exception in this case? I guess we're fixing the problem at the call site below?

Comment (HEskandari) [inline src/ServiceInsight/ServiceControl/ServiceControlConnectionProvider.cs]: Don't think so. This is called on application start as well which means if an exception is thrown the app will crash.

**Pull request merged**